Realm/RLMRealm.mm

```diff
@@ -766,6 +766,32 @@ + (NSUInteger)schemaVersionAtPath:(NSString *)realmPath encryptionKey:(NSData *)
     return RLMRealmSchemaVersion(realm);
 }
 
++ (BOOL)migrationRequiredAtPath:(NSString *)realmPath encryptionKey:(NSData *)key error:(NSError **)outError {
+    RLMRealm *realm = RLMGetThreadLocalCachedRealmForPath(realmPath);
```

Comment:
It only reads from the realm if it opened it. This part is only checking if the realm is already open, and if it is it returns without doing anything further.